#!/usr/bin/env python
#
# Copyright (c) 2016, Adam Allevato
# Copyright (c) 2017, The University of Texas at Austin
# All rights reserved.
#
# Redistribution and use in source and binary forms, with or without
# modification, are permitted provided that the following conditions are met:
#
# 1. Redistributions of source code must retain the above copyright notice,
#    this list of conditions and the following disclaimer.
#
# 2. Redistributions in binary form must reproduce the above copyright
#    notice, this list of conditions and the following disclaimer in the
#    documentation and/or other materials provided with the distribution.
#
# 3. Neither the name of the copyright holder nor the names of its
#    contributors may be used to endorse or promote products derived from
#    this software without specific prior written permission.
#
# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
# IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
# ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

"""
Dynamic reconfigure configuration for the cylinder classifier node. These
parameters are passed on to the node so it can configure itself accordingly.
"""

PACKAGE = "orp"
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()
##############################################################################
# function signature:
# gen.add("name", type(double_t|bool_t|int_t), 0, desc, default, min, max)

# BASIC TUNING PARAMETERS
# TODO(Kukanani): add descriptions for each
gen.add("normal_distance_weight", double_t, 0, "", 0.1, 0, 1)
gen.add("max_iterations", double_t, 0, "", 10000, 1, 1000000)
gen.add("distance_threshold", double_t, 0, "", 0.05, 0, 0.5)
gen.add("min_radius", double_t, 0, "", 0.0005, 0.0005, 0.1)
gen.add("max_radius", double_t, 0, "", 0.1, 0.0025, 0.3)

##############################################################################
exit(gen.generate(PACKAGE, "orp", "CylinderClassifier"))
